### DISCUSSION TITLE
Issues with Zero-Gas Transactions on Subnet-EVM

### DISCUSSION THREAD

**Discussion by furkan-ux** [ava-labs/subnet-evm#785] [General]
Hello everyone,

We're facing an issue on the Subnet-EVM where transactions with no gas aren't being added to the block. From my previous experience, I recall EIP-1559 being the culprit, and tweaking the genesis doesn't seem to alleviate the issue; these transactions just remain in the mempool.

Having the capability to process 0-gas transactions is crucial for various applications. This is especially true for enterprise and/or private networks, and chains that employ the EVM for its simplicity but introduce unique account structures. Some privacy-centric frameworks, for example, necessitate the use of throwaway accounts (generated through random private keys) to sign transactions. This strategy is implemented to counteract gas-linkability issue. Generally speaking, in these frameworks, account IDs are included in the transaction data instead of the conventional msg.sender.

While I acknowledge the potential security risks (like DoS attacks) associated with enabling 0-gas transactions, we (EtraPay) have other security measures in place to mitigate these risks.

Any guidance on how to seamlessly enable this feature would be greatly appreciated.

Thank you!

**Comment by aaronbuchwald** [comment]
Hey @furkan-ux , you're right 0-gas transactions are supported by Ethereum, so I'd be in favor of making sure that they are supported in Subnet-EVM and deferring to the Subnet maintainer to ensure that sufficient DoS protection mechanisms are in place elsewhere.
